pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -272,15 +302,19 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 			glog.V(4).Infof("synchronizing unbound PersistentVolumeClaim[%s]: no volume found", claimToClaimKey(claim))
 			// No PV could be found
 			// OBSERVATION: pvc is "Pending", will retry
-			if v1helper.GetPersistentVolumeClaimClass(claim) != "" {
+			if v1helper.GetPersistentVolumeClaimClass(claim) != "" && !delayBinding {
```

Comment:
reordering avoids the negative case:
```go
switch {
case delayBinding:
...
case v1helper.GetPersistentVolumeClaimClass(claim) != "":
...
default:
...
}
```

Reply:
done.  @jsafrane can you also take a look at this update? i squashed it into the 3rd commit.

Reply:
checked pv_controller.go,  lgtm